sway-core/src/semantic_analysis/namespace/trait_map.rs

```diff
@@ -1377,4 +1420,27 @@ impl TraitMap {
             Ok(())
         })
     }
+
+    fn get_impls_mut(&mut self, engines: &Engines, type_id: TypeId) -> &mut im::Vector<TraitEntry> {
+        let type_str = (*engines.te().get(type_id)).get_type_str_filter(engines);
+        if !self.trait_impls.contains_key(&type_str) {
+            self.trait_impls.insert(type_str.clone(), im::Vector::new());
+        }
+
+        self.trait_impls.get_mut(&type_str).unwrap()
+    }
+
+    fn get_impls(&self, engines: &Engines, type_id: TypeId) -> im::Vector<TraitEntry> {
+        let type_str = (*engines.te().get(type_id)).get_type_str_filter(engines);
+        let mut vec = self.trait_impls.get(&type_str).cloned().unwrap_or_default();
+        if type_str != *"_" {
```

Comment:
Never saw `*"_"`. Why it is needed?

Reply:
Equivalent to `&type_str != "_"` I assume.

Reply:
That was my point... maybe that makes more sense... :D